lib/tests/streamlit/elements/lib/column_config_utils_test.py

```diff
@@ -0,0 +1,329 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import datetime
+import unittest
+from decimal import Decimal
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+from parameterized import parameterized
+
+from streamlit.elements.lib.column_config_utils import (
+    ColumnDataKind,
+    _determine_data_kind,
+    _determine_data_kind_via_arrow,
+    _determine_data_kind_via_inferred_type,
+    _determine_data_kind_via_pandas_dtype,
+    determine_dataframe_schema,
+)
+
+
+class TestObject(object):
+    def __str__(self):
+        return "TestObject"
+
+
+def _get_arrow_schema_field(column: pd.Series) -> pa.Field | None:
+    """Get the Arrow schema field for a pandas Series."""
+    try:
+        arrow_schema = pa.Table.from_pandas(column.to_frame()).schema
+        return arrow_schema.field(0)
+    except (pa.ArrowTypeError, pa.ArrowInvalid, pa.ArrowNotImplementedError):
+        return None
+
+
+SHARED_DATA_KIND_TEST_CASES = [
+    (pd.Series(["a", "b", "c"], dtype=pd.StringDtype()), ColumnDataKind.STRING),
+    # We need to use Int64 here, otherwise it gets converted to float if a None is added:
+    (pd.Series([1, 2, 3], dtype="Int64"), ColumnDataKind.INTEGER),
```

Comment:
Should we maybe add a negative integer here? Maybe a negative float too?

Reply:
Good point, I added both :)